coderd/database/lock.go

```diff
@@ -12,6 +12,8 @@ const (
 	LockIDDBPurge
 	LockIDNotificationsReportGenerator
 	LockIDCryptoKeyRotation
+	LockIDReconcileTemplatePrebuilds
```

Comment:
We need two separate locks because we'll have a [wider reconciliation loop](https://github.com/coder/coder/blob/dk/prebuilds/enterprise/coderd/prebuilds/reconcile.go#L213) which must hold a global lock, and an inner lock which is used to protect the piece where we [determine the state](https://github.com/coder/coder/blob/dk/prebuilds/enterprise/coderd/prebuilds/reconcile.go#L242). The latter can be called outside of a reconciliation loop ([as in the metrics collection](https://github.com/coder/coder/blob/dk/prebuilds/enterprise/coderd/prebuilds/metricscollector.go#L68)).